### DISCUSSION TITLE
mAP for object detection on test split

### DISCUSSION THREAD

**Discussion by AashishKumbashi** [azavea/raster-vision#2025] [Q&A] [answered]
I understand that after every epoch, rastervision helps in calculating the mAP on the validation dataset. However, later when analysing the predictions, I have only precision, recall and F1 score.

Is there a way to somehow obtain the mAP scores for the test dataset?

**Comment by AdeelH** [comment, marked as answer]
You are right. That's some missing functionality. To fix that, we would need to adapt the mAP calculation code from [here](https://github.com/azavea/raster-vision/blob/0.21/rastervision_pytorch_learner/rastervision/pytorch_learner/object_detection_utils.py#L79) to [here](https://github.com/azavea/raster-vision/blob/0.21/rastervision_core/rastervision/core/evaluation/object_detection_evaluation.py).

A workaround might be to instantiate a `Learner` with the test dataset and the trained weights and then call `Learner.eval_model('test')`.
